st3/lsp_utils/npm_client_handler.py

```diff
@@ -21,39 +26,33 @@ def is_node_installed():
 
 
 class ApiWrapper(object):
-    def __init__(self, client):
-        self.__client = client
+    def __init__(self, plugin: AbstractPlugin):
+        self.__plugin = plugin
 
     def on_notification(self, method: str, handler: Callable) -> None:
-        self.__client.on_notification(method, handler)
+        setattr(self.__plugin, method2attr(method), lambda this, params: handler(params))
 
     def on_request(self, method: str, handler: Callable) -> None:
-        def on_response(params, request_id):
-            handler(params, lambda result: send_response(request_id, result))
 
         def send_response(request_id, result):
-            self.__client.send_response(Response(request_id, result))
+            session = self.__plugin.weaksession()
+            if session:
+                session.send_response(Response(request_id, result))
+
+        def on_response(this, params, request_id):
+            handler(params, lambda result: send_response(request_id, result))
 
-        self.__client.on_request(method, on_response)
+        setattr(self.__plugin, method2attr(method), on_response)
```

Comment:
Can't we hide this behind an API? Why do we need to manually fiddle with `method2attr` function? Could just create `AbstractPlugin.on_request` and `AbstractPlugin.on_notification` that would do that behind the scenes.

Reply:
The NpmClientHandler is the API that hides it?

Reply:
I'd like this hidden from public API. I imagine some packages will not use `NpmClientHandler`.